sw/device/tests/example_test_from_flash.c

```diff
@@ -73,10 +73,15 @@ const test_config_t kTestConfig = {
 /**
  * Place data in flash that will need to persist across resets by marking with
  * with the section ".non_volatile_scratch"). Write to this region with the
- * flash controller DIF. Read from it with `abs_mmio_read32(...)`.
+ * flash controller DIFs. Read from it with `abs_mmio_read32(...)`.
+ * Additionally, be sure to initialize the bits in this array to all 1s,
+ * otherwise this region may not be programmed properly, depending on the
+ * programming transactions issued. According to the hardware specification:
+ *  - a bit cannot be programmed back to 1 once it has been programmed to 0, and
+ *  - only erase can restore a bit to 1 under normal circumstances.
  */
 // __attribute__((section(".non_volatile_scratch")))
-// const volatile uint32_t non_volatile_data[SIZE_OF_DATA];
+// const volatile uint32_t non_volatile_data[5] = {-1, -1, -1, -1, -1};
```

Comment:
nit: Probably ought to use `UINT32_MAX` for the correct sign for the literal.

Reply:
ooo right, thanks. fixed.